execution_engine_testing/test_support/src/transfer.rs

```diff
@@ -0,0 +1,287 @@
+                runtime_args! { ARG_TARGET_PURSE => purse, ARG_AMOUNT => U512::one() },
+            )
+            .with_authorization_keys(&[TARGET_ADDR])
+            .with_deploy_hash(make_deploy_hash(i)) // deploy_hash
+            .build();
+        exec_builder = exec_builder.push_deploy(deploy);
+    }
+
+    let exec_request = exec_builder.build();
+
+    let builder = builder.exec(exec_request).expect_success();
+    if should_commit {
+        builder.commit();
+    }
+}
+
+/// This test simulates flushing at the end of a block.
+pub fn transfer_to_account_multiple_native_transfers(
+    builder: &mut LmdbWasmTestBuilder,
+    execute_requests: &[ExecuteRequest],
+    use_scratch: bool,
+) {
+    for exec_request in execute_requests {
+        let request = ExecuteRequest::new(
+            exec_request.parent_state_hash,
+            exec_request.block_time,
+            exec_request.deploys.clone(),
+            exec_request.protocol_version,
+            exec_request.proposer.clone(),
+        );
+        if use_scratch {
+            builder.scratch_exec_and_commit(request).expect_success();
+        } else {
+            builder.exec(request).expect_success();
+            builder.commit();
+        }
+    }
+    if use_scratch {
+        builder.write_scratch_to_lmdb();
+    }
+    // flush to disk only after entire block (simulates manual_sync_enabled=true config entry)
+    builder.flush_environment();
+
+    // WasmTestBuilder holds on to all execution results. This needs to be cleared to reduce
+    // overhead in this test - it will likely OOM without.
+    builder.clear_results();
+}
+
+/// Generate many native transfers from target_account.
+pub fn create_multiple_native_transfers_to_purses(
+    source_account: AccountHash,
+    transfer_count: usize,
+    purses: &[URef],
+) -> Vec<ExecuteRequest> {
+    let mut purse_index = 0usize;
+    let mut exec_requests = Vec::with_capacity(transfer_count);
+    for _ in 0..transfer_count {
+        let account = {
+            let account = purses[purse_index];
+            if purse_index == purses.len() - 1 {
+                purse_index = 0;
+            } else {
+                purse_index += 1;
+            }
+            account
+        };
+        let mut exec_builder = ExecuteRequestBuilder::new();
+        let runtime_args = runtime_args! {
+            ARG_TARGET => account,
+            ARG_AMOUNT => U512::one(),
+            ARG_ID => <Option<u64>>::None
+        };
+        let native_transfer = DeployItemBuilder::new()
+            .with_address(source_account)
+            .with_empty_payment_bytes(runtime_args! {})
+            .with_transfer_args(runtime_args)
+            .with_authorization_keys(&[source_account])
+            .build();
+        exec_builder = exec_builder.push_deploy(native_transfer);
+        let exec_request = exec_builder.build();
+        exec_requests.push(exec_request);
+    }
+    exec_requests
+}
```

Comment:
I have a vague memory of seeing it somewhere already but I don't see this code being deleted anywhere. Is this a new code or a copied one?